test/e2e/monitoring_stack_controller_test.go

```diff
@@ -568,15 +570,28 @@ func newAlerts(t *testing.T) *monv1.PrometheusRule {
 	return rule
 }
 
-func newMonitoringStack(t *testing.T, name string, options ...func(*stack.MonitoringStack)) *stack.MonitoringStack {
+type stackMod func(*stack.MonitoringStack)
```

Comment:
Nit: I think a comment on how this is used might be nice or a more explicit name? `stackModifier` would already do it for me tbh.

Reply:
:) agreed. Will change this